uni/lib/controller/parsers/parser_exams.dart

```diff
@@ -51,13 +51,19 @@ class ParserExams {
                     .queryParameters['p_exa_id']!;
               }
               if (examsDay.querySelector('span.exame-sala') != null) {
-                rooms =
-                    examsDay.querySelector('span.exame-sala')!.text.split(',');
+                rooms = examsDay
+                    .querySelector('span.exame-sala')!
+                    .text
+                    .split(',')
+                    .map((e) => e.trim())
+                    .toList();
               }
-              schedule = examsDay.text.substring(
-                examsDay.text.indexOf(':') - 2,
-                examsDay.text.indexOf(':') + 9,
-              );
+              schedule = examsDay.text.endsWith('-')
+                  ? '00:00-00:00'
+                  : examsDay.text.substring(
+                      examsDay.text.indexOf(':') - 2,
+                      examsDay.text.indexOf(':') + 9,
+                    );
```

Comment:
Can you try to improve the original code, removing the hardcoded numbers? Say, split before and after the `:`, and tryParse it.

Reply:
I wish it were possible, but some of the course units have numbers in them, making it not possible to split only by `:`. The exam fetcher provides a single string with all the exam information (ex. `FIS20190:00-`).

Reply:
The subject is in a `b` tag as far as I can see. Wouldn't that help?